internal/listener/listener.go

```diff
@@ -447,7 +450,14 @@ func (l *Listener) ProcessEvent(w http.ResponseWriter, req *http.Request) {
 			escalationRecipients[escalationRecipient.Key] = true
 
 			if !managed || state.Role > incident.RoleRecipient {
-				for _, c := range escalationRecipient.Recipient.GetContactsAt(ev.Time) {
+				r := escalationRecipient.Recipient
+				_, isContact := r.(*recipient.Contact)
+				contacts := r.GetContactsAt(ev.Time)
+				for _, c := range contacts {
+					if !isContact {
+						groupsOrSchedules[c] = recipient.ToKey(r)
+					}
```

Comment:
What's supposed to happen if the same contact if references multiple times, i.e. directly, via a group and via a schedule? This would just arbitrarily pick a group or schedule. We could write all of that information into the database, but that would probably mean writing multiple rows for the same notification, i.e. the web interface would have to handle and display this probably, so I don't want to decide this on my own.

No matter how we do it, a comment in the schema file how these rows are to be interpreted would be great.